From 05e2d17204fcabd2e23b29345a9687bc14a830b9 Mon Sep 17 00:00:00 2001
From: Xylar Asay-Davis <xylarstorm@gmail.com>
Date: Tue, 12 Nov 2019 10:54:57 +0100
Subject: [PATCH] Add missing MPI_Finalize to imoab_coupler test

---
 test/parallel/imoab_coupler.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/test/parallel/imoab_coupler.cpp b/test/parallel/imoab_coupler.cpp
index 230991ec8..a305e96e5 100644
--- a/test/parallel/imoab_coupler.cpp
+++ b/test/parallel/imoab_coupler.cpp
@@ -470,6 +470,7 @@ int main( int argc, char* argv[] )
   MPI_Group_free(&jgroup);
   MPI_Comm_free(&jcomm);
 
+  MPI_Finalize();
   return 0;
 }
 
-- 
2.23.0

